CMakeLists.txt

```diff
@@ -835,32 +836,18 @@ set (HDF5_SRC_INCLUDE_DIRS
     ${HDF5_SRC_INCLUDE_DIRS}
     ${H5FD_SUBFILING_DIR}
 )
-option (HDF5_ENABLE_SUBFILING_VFD "Build Parallel HDF5 Subfiling VFD" OFF)
+cmake_dependent_option(HDF5_ENABLE_SUBFILING_VFD "Build Parallel HDF5 Subfiling VFD" ON "HDF5_ENABLE_PARALLEL AND NOT WIN32" OFF)
 if (HDF5_ENABLE_SUBFILING_VFD)
-  if (WIN32)
```

Comment:
I believe this check is still needed. `cmake_dependent_option` just hides the option, but someone could still set it on Win32.

Reply:
I'd like to know for sure - as I believe the documentation states:
<condition>: A boolean expression involving other CMake options. It must evaluate to true or false.
Of course NOT WIN32 might not evaluate correctly.

Reply:
What I mean is that CMake hides the option, but I don't think that means it's actually gone. I think it's similar to just being marked with `mark_as_advanced()`.

Reply:
Yes it hides the option but the condition forces a value if it is hidden

Reply:
I ran " ccmake -DMPIEXEC_PREFLAGS:STRING="-n;6" -DHDF5_ENABLE_SUBFILING_VFD=ON ../lrk-hdf5", then configured without any other changes.  The subfiling vfd option was missing and libhdf5.settings reported Subfiling VFD  OFF.